test/cctest/node_test_fixture.h

```diff
@@ -53,72 +53,61 @@ struct Argv {
   int nr_args_;
 };
 
-extern uv_loop_t current_loop;
 
 class NodeTestFixture : public ::testing::Test {
- public:
-  static uv_loop_t* CurrentLoop() { return &current_loop; }
-
-  node::MultiIsolatePlatform* Platform() const { return platform_; }
-
  protected:
+  static std::unique_ptr<v8::ArrayBuffer::Allocator> allocator;
+  static std::unique_ptr<v8::TracingController> tracing_controller;
+  static std::unique_ptr<node::NodePlatform> platform;
+  static v8::Isolate::CreateParams params;
+  static uv_loop_t current_loop;
   v8::Isolate* isolate_;
 
-  ~NodeTestFixture() {
-    TearDown();
-  }
-
-  virtual void SetUp() {
+  static void SetUpTestCase() {
+    platform.reset(new node::NodePlatform(8, nullptr));
+    tracing_controller.reset(new v8::TracingController());
+    allocator.reset(v8::ArrayBuffer::Allocator::NewDefaultAllocator());
+    params.array_buffer_allocator = allocator.get();
+    node::tracing::TraceEventHelper::SetTracingController(
+        tracing_controller.get());
     CHECK_EQ(0, uv_loop_init(&current_loop));
-    platform_ = new node::NodePlatform(8, nullptr);
-    v8::V8::InitializePlatform(platform_);
+    v8::V8::InitializePlatform(platform.get());
     v8::V8::Initialize();
-    v8::Isolate::CreateParams params_;
-    params_.array_buffer_allocator = allocator_.get();
-    isolate_ = v8::Isolate::New(params_);
-
-    // As the TracingController is stored globally, we only need to create it
-    // one time for all tests.
-    if (node::tracing::TraceEventHelper::GetTracingController() == nullptr) {
-      node::tracing::TraceEventHelper::SetTracingController(
-          new v8::TracingController());
-    }
   }
 
-  virtual void TearDown() {
-    if (platform_ == nullptr) return;
-    platform_->Shutdown();
+  static void TearDownTestCase() {
+    platform->Shutdown();
     while (uv_loop_alive(&current_loop)) {
       uv_run(&current_loop, UV_RUN_ONCE);
     }
     v8::V8::ShutdownPlatform();
-    delete platform_;
-    platform_ = nullptr;
     CHECK_EQ(0, uv_loop_close(&current_loop));
   }
 
- private:
-  node::NodePlatform* platform_ = nullptr;
-  std::unique_ptr<v8::ArrayBuffer::Allocator> allocator_{
-      v8::ArrayBuffer::Allocator::NewDefaultAllocator()};
+  virtual void SetUp() {
+    isolate_ = v8::Isolate::New(params);
```

Comment:
Can you add a `CHECK_NE(isolate_, nullptr)`?

Reply:
Sure thing, I'll add the check. Thanks